java/source/demo/pom.xml

```diff
@@ -27,13 +25,11 @@
       </extension>
     </extensions>
     </build>
-
     <properties>
-        <maven.compiler.source>8</maven.compiler.source>
-        <maven.compiler.target>8</maven.compiler.target>
-        <arrow.version>9.0.0</arrow.version>
+        <maven.compiler.source>injected_by_ci</maven.compiler.source>
```

Comment:
I would still much rather these have a default version so that it can be easily used locally.